### ISSUE TITLE
Resume training from checkpoint

### ISSUE THREAD

**Issue by thaihawai** [chs20/RobustVLM#22]
Hello. I want to ask how can I resume training from checkpoint? I see that you save a lot of checkpoint during training but there is no document on how to resume training.

**Comment by chs20** [comment]
Hi, something like this should work in combination with the usual training parameters:
```
--optimizer_state /path/to/fallback_600_opt.pt --start_step 600
```

**Comment by thaihawai** [comment]
Thanks you very much. Also when training I often see the eval acc goes down when the racc stay mostly small. Is this expected behavior? I'm asking this because so far I've only been training on ~1000/20000 steps so I dont know if this is correct or something is wrong.

This is the training param I'm using - **ViT-B/32 with batch size 64**
`python -m train.adversarial_training_clip --clip_model_name ViT-B-32 --pretrained openai --dataset imagenet --imagenet_root ../data/ImageNet --template std --output_normalize False --steps 20000 --warmup 1400 --batch_size 64 --loss l2 --opt adamw --lr 1e-5 --wd 1e-4 --attack pgd --inner_loss l2 --norm linf --eps 4 --iterations_adv 10 --stepsize_adv 1 --wandb False --ou
tput_dir ./outdir/ --experiment_name FARE4 --log_freq 10 --eval_freq 10 `

![Image](https://github.com/user-attachments/assets/130e8e0b-f468-4f4c-a300-e844345c1552)

**Comment by chs20** [comment]
That looks fine, the acc should slowly recover and the racc should slowly increase, but it takes time. Here is a log from a ViT-B/32 Openai run (x-axis is step/10, so total 20k steps, eps=4):

![Image](https://github.com/user-attachments/assets/a0e98b36-4a02-47bc-b595-252c55c11364)

**Comment by thaihawai** [comment]
I guess that's fine then. I see the acc drop too much but racc does not improve and got panic. Thank you.
